stream_alert_cli/test.py

```diff
@@ -48,338 +56,486 @@ class TestingSuppressFilter(logging.Filter):
-                                                                     failure[0], failure[1],
-                                                                     color)
-
-    # Check if there were any warnings and report on them
-    if rules_fail_pass[2]:
-        color = COLOR_YELLOW
-        warning_count = len(rules_fail_pass[2])
-        print '{}{} \tWarning{}'.format(color, warning_count, ('', 's')[warning_count > 1])
-
-        for index, failure in enumerate(rules_fail_pass[2]):
-            if index == warning_count-1:
-                # Change the color back so std out is not yellow
-                color = COLOR_RESET
-            print '\t({}/{}) {} [{}]{}'.format(index+1, warning_count, failure[1],
-                                               failure[0], color)
-
-def test_rule(rule_name, test_record, formatted_record):
-    """Feed formatted records into StreamAlert and check for alerts
-    Args:
-        rule_name: The rule name being tested
-        test_record: A single record to test
-        formatted_record: A properly formatted version of record for the service to be tested
-
-    Returns:
-        boolean indicating if this rule passed
-    """
-    event = {'Records': [formatted_record]}
-
-    trigger_count = test_record.get('trigger_count')
-    if trigger_count:
-        expected_alert_count = trigger_count
-    else:
-        expected_alert_count = (0, 1)[test_record['trigger']]
 
-    # Start mocked sns
-    BOTO_MOCKER_SNS.start()
-
-    # Create the topic used for the mocking of alert sending
-    boto3.client('sns', region_name='us-east-1').create_topic(Name='test_streamalerts')
+class RuleProcessorTester(object):
+    """Class to encapsulate testing the rule processor"""
+
+    def __init__(self, print_output):
+        super(self.__class__, self).__init__()
+        # Create the topic used for the mocking of alert sending
+        # This is used in stream_alert/rule_processor/sink.py to 'send' alerts
+        sns_client = boto3.client('sns', region_name='us-east-1')
+        sns_client.create_topic(Name='test_streamalerts')
+        # Create a list for pass/fails. The first value in the list is a
+        # list of tuples for failures, and the second is list of tuples for
+        # passes. Tuple is (rule_name, rule_description)
+        self.rules_fail_pass_warn = ([], [], [])
+        self.print_output = print_output
+
+    def test_processor(self, rules):
+        """Perform integration tests for the 'rule' Lambda function
+
+        Returns:
+            [generator] yields a tuple containig a boolean of test status and
+                a list of alerts to run through the alert processor
+        """
+        all_tests_passed = True
+
+        for rule_file, rule_name in get_rule_test_files(rules):
+            with open(os.path.join(DIR_RULES, rule_file), 'r') as rule_file_handle:
+                try:
+                    contents = json.load(rule_file_handle)
+                except Exception as err:
```

Comment:
From the docs:

```If the data being deserialized is not a valid JSON document, a ValueError will be raised.```

Our error description does not match the type of Exceptions we are trying to catch.  Either make it more general and add `TypeError` and `OverflowError`, or just catch `ValueError` and call it a day